service/history/workflow/mutable_state_impl_restart_activity_test.go

```diff
@@ -0,0 +1,370 @@
+package workflow
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/golang/mock/gomock"
+	"github.com/pborman/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"github.com/uber-go/tally/v4"
+	commandpb "go.temporal.io/api/command/v1"
+	commonpb "go.temporal.io/api/common/v1"
+	enumspb "go.temporal.io/api/enums/v1"
+	failurepb "go.temporal.io/api/failure/v1"
+	taskqueuepb "go.temporal.io/api/taskqueue/v1"
+	enumsspb "go.temporal.io/server/api/enums/v1"
+	persistencespb "go.temporal.io/server/api/persistence/v1"
+	"go.temporal.io/server/common/backoff"
+	commonclock "go.temporal.io/server/common/clock"
+	"go.temporal.io/server/common/log"
+	"go.temporal.io/server/common/primitives/timestamp"
+	"go.temporal.io/server/service/history/configs"
+	"go.temporal.io/server/service/history/events"
+	"go.temporal.io/server/service/history/shard"
+	"go.temporal.io/server/service/history/tests"
+	"google.golang.org/protobuf/types/known/durationpb"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type (
+	BackoffIntervalCalculator = func(
+		time.Time,
+		int32,
+		int32,
+		*durationpb.Duration,
+		*durationpb.Duration,
+		*timestamppb.Timestamp,
+		float64,
+		*failurepb.Failure,
+		[]string,
+	) (time.Duration, enumspb.RetryState)
+
+	snapshot struct {
+		mutableStateApproximateSize int
+		activitySize                int
+	}
+
+	retrySuite struct {
+		suite.Suite
+		require *require.Assertions
+		assert  *assert.Assertions
```

Comment:
You don't need both require and assert, and you aren't using require at all. You don't even need to embed assert in the first place as the [`suite.Suite`](https://pkg.go.dev/github.com/stretchr/testify/suite#Suite)already does.

If you want assertions that won't immediately fail your test all you need is:

```
type retrySuite struct {
    suite.Suite
}

// normal initialization
func (s *retrySuite) TestFooBarBaz() {
    s.True(false)
    s.True(true)
}
```

If you want all your assertions to immediately fail the test _that's_ when we embed the `require.Assertions` type:

```
type retrySuite struct {
    suite.Suite
    *require.Assertions
}

// normal initialization

func (s *retrySuite) TestFooBarBaz() {
    s.True(false) // The test will fail and exit here
    s.True(true)
}
```